### ISSUE TITLE
Streaming WebRTC from Encoder Output

### ISSUE THREAD

**Issue by leoRl** [sourcey/libsourcey#172]
Hi !
I'm starting to test LibSourcey to play video files on a web browser.
Just a quick question, is there any way to retrieve an encoder stream output in order to stream it through webrtc ?
I've only seen a file input option in "MultiplexMediaCapturer". 

Thanks in advance.

**Comment by auscaster** [comment]
Yes indeed! You can swap the internal MediaCapturer for a MultiplexEncoder
and you're off to the races :)

**Comment by leoRl** [comment]
Thanks for the quick answer, another thing that i don't understand, why h264 doesn't appear in the offer sdp with the example webrtcstreamer ? 

`"sdp":"v=0\r\no=- 5416046408381757456 2 IN IP4 127.0.0.1\r\ns=-\r\nt=0 0\r\na=group:BUNDLE audio video\r\na=msid-semantic: WMS stream_label\r\nm=audio 9 UDP/TLS/RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 110 112 113 126\r\nc=IN IP4 0.0.0.0\r\na=rtcp:9 IN IP4 0.0.0.0\r\na=ice-ufrag:5jey\r\na=ice-pwd:2BlFdv35K0TKy3Ost+zkFEKg\r\na=fingerprint:sha-256 2B:EE:18:E0:A4:22:E7:79:C5:6F:4D:08:2E:AA:D6:FA:78:67:59:1C:BB:56:0F:9D:57:C3:AB:95:92:73:3A:94\r\na=setup:actpass\r\na=mid:audio\r\na=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level\r\na=sendrecv\r\na=rtcp-mux\r\na=rtpmap:111 opus/48000/2\r\na=rtcp-fb:111 transport-cc\r\na=fmtp:111 minptime=10;useinbandfec=1\r\na=rtpmap:103 ISAC/16000\r\na=rtpmap:104 ISAC/32000\r\na=rtpmap:9 G722/8000\r\na=rtpmap:102 ILBC/8000\r\na=rtpmap:0 PCMU/8000\r\na=rtpmap:8 PCMA/8000\r\na=rtpmap:106 CN/32000\r\na=rtpmap:105 CN/16000\r\na=rtpmap:13 CN/8000\r\na=rtpmap:110 telephone-event/48000\r\na=rtpmap:112 telephone-event/32000\r\na=rtpmap:113 telephone-event/16000\r\na=rtpmap:126 telephone-event/8000\r\na=ssrc:1316800838 cname:ZWHjbagyMOQ/xo1h\r\na=ssrc:1316800838 msid:stream_label audio_label\r\na=ssrc:1316800838 mslabel:stream_label\r\na=ssrc:1316800838 label:audio_label\r\nm=video 9 UDP/TLS/RTP/SAVPF 96 98 100 127 97 99 101\r\nc=IN IP4 0.0.0.0\r\na=rtcp:9 IN IP4 0.0.0.0\r\na=ice-ufrag:5jey\r\na=ice-pwd:2BlFdv35K0TKy3Ost+zkFEKg\r\na=fingerprint:sha-256 2B:EE:18:E0:A4:22:E7:79:C5:6F:4D:08:2E:AA:D6:FA:78:67:59:1C:BB:56:0F:9D:57:C3:AB:95:92:73:3A:94\r\na=setup:actpass\r\na=mid:video\r\na=extmap:2 urn:ietf:params:rtp-hdrext:toffset\r\na=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\na=extmap:4 urn:3gpp:video-orientation\r\na=extmap:5 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01\r\na=extmap:6 http://www.webrtc.org/experiments/rtp-hdrext/playout-delay\r\na=sendrecv\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:96 VP8/90000\r\na=rtcp-fb:96 ccm fir\r\na=rtcp-fb:96 nack\r\na=rtcp-fb:96 nack pli\r\na=rtcp-fb:96 goog-remb\r\na=rtcp-fb:96 transport-cc\r\na=rtpmap:98 VP9/90000\r\na=rtcp-fb:98 ccm fir\r\na=rtcp-fb:98 nack\r\na=rtcp-fb:98 nack pli\r\na=rtcp-fb:98 goog-remb\r\na=rtcp-fb:98 transport-cc\r\na=rtpmap:100 red/90000\r\na=rtpmap:127 ulpfec/90000\r\na=rtpmap:97 rtx/90000\r\na=fmtp:97 apt=96\r\na=rtpmap:99 rtx/90000\r\na=fmtp:99 apt=98\r\na=rtpmap:101 rtx/90000\r\na=fmtp:101 apt=100\r\na=ssrc-group:FID 3027468101 2863773322\r\na=ssrc:3027468101 cname:ZWHjbagyMOQ/xo1h\r\na=ssrc:3027468101 msid:stream_label video_label\r\na=ssrc:3027468101 mslabel:stream_label\r\na=ssrc:3027468101 label:video_label\r\na=ssrc:2863773322 cname:ZWHjbagyMOQ/xo1h\r\na=ssrc:2863773322 msid:stream_label video_label\r\na=ssrc:2863773322 mslabel:stream_label\r\na=ssrc:2863773322 label:video_label\r\n","type":"offer"}]
["symple:webrtc: recv remote candiate",{"candidate":"candidate:4112619227 1 udp 2122260223 172.17.0.2 45476 typ host generation 0 ufrag 5jey network-id 1 network-cost 50","sdpMLineIndex":0,"sdpMid":"audio"}]`

**Comment by auscaster** [comment]
I think you need to compile WebRTC with h.264 support. Some info on this
thread: https://groups.google.com/forum/#!topic/discuss-webrtc/ZWrebkiWvo0

**Comment by auscaster** [comment]
Did you make any progress here?

**Comment by leoRl** [comment]
Sorry for the lake of response.

Well I still have some difficulties to understand fully the architecture of the PacketStream class, specially with the output part. 
What I've done is that I attached to the packetStream a MultiplexPacketEncoder with a mxf file as source. I linked the encoder with some options given by the class EncoderOptions.
Then for the WebRTC part, I  added a track with the packetStream.emitter as source.
What is missing here ? something to synchronize the packets ? 

Thanks in advance.
